### PR TITLE
strategy: add harmonic shark pattern recognition

### PR DESCRIPTION
**Pull request by austin362667**

### CODE REVIEW

File: pkg/strategy/harmonic/strategy.go [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+	positionLock        sync.RWMutex `ignore:"true"`
+	startTime           time.Time
+	minutesCounter      int
+	orderPendingCounter map[uint64]int
+	frameKLine          *types.KLine
+	kline1m             *types.KLine
+
+	beta float64
+
+	StopLoss                  fixedpoint.Value `json:"stoploss"`
```
Comment (c9s) [inline pkg/strategy/harmonic/strategy.go:75]: not used ?

File: pkg/strategy/harmonic/strategy.go [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+	CanvasPath                string           `json:"canvasPath"`
+	PredictOffset             int              `json:"predictOffset"`
+	HighLowVarianceMultiplier float64          `json:"hlVarianceMultiplier"`
```
Comment (c9s) [inline pkg/strategy/harmonic/strategy.go:78]: not used?

**Review by andycheng123 — APPROVED** [review]

**Review by c9s — APPROVED** [review]

File: pkg/strategy/harmonic/strategy.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+		log.WithError(err).Errorf("cannot render in harmonic")
+	}
+
+	canvas = s.DrawPNL(profit)
+	f, err = os.Create(s.GraphPNLPath)
+	if err != nil {
+		log.WithError(err).Errorf("open pnl")
+		return
+	}
+	defer f.Close()
```
Comment (c9s) [inline pkg/strategy/harmonic/strategy.go:354]: the f here is in defer, and f will be overridden.

File: pkg/strategy/harmonic/draw.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	defer f.Close()
+	if err = canvas.Render(chart.PNG, f); err != nil {
+		return fmt.Errorf("cannot render pnl")
+	}
+	canvas = DrawCumPNL(s.InstanceID(), cumProfit)
+	f, err = os.Create(s.GraphCumPNLPath)
+	if err != nil {
+		return fmt.Errorf("cannot create on path " + s.GraphCumPNLPath)
+	}
+	defer f.Close()
```
Comment (c9s) [inline pkg/strategy/harmonic/draw.go:53]: the f here can be overridden

**Review by c9s — APPROVED** [review]

**Pull request merged**